x/core/keeper/gas_meter.go

```diff
@@ -63,9 +62,82 @@ func (k Keeper) DistributionsFromGasMeter(ctx sdk.Context, reqID string, reqHeig
 	return dists
 }
 
+func (k Keeper) ProcessDistributions(ctx sdk.Context, dists []types.Distribution, dr *types.DataRequest, minimumStake math.Int) error {
+	// Distribute in order.
+	denom, err := k.stakingKeeper.BondDenom(ctx)
+	if err != nil {
+		return err
+	}
+
+	remainingEscrow := dr.Escrow
+
+	// TODO Events
+	var amount math.Int
+	for _, dist := range dists {
```

Comment:
Missing the check that the escrow didn't run out before looking at the cases. In that case we should break the loop early without processing the rest of the messages.